### PR TITLE
chore: rely on be to trigger face auth

### PR DESCRIPTION
**Pull request by ice-hades**

### CODE REVIEW

File: src/store/modules/Tokenomics/sagas/startMiningSession.ts [hunk 1]
```diff
@@ -125,6 +103,30 @@ export function* startMiningSessionSaga(
...
+        });
+        return;
+      }
+    } else if (isApiError(error, 403, 'MINING_DISABLED')) {
+      const errorData = error?.response?.data?.data;
+      if (errorData && typeof errorData.kycStepBlocked === 'number') {
+        yield removeScreenByName('Tooltip').catch();
+        navigate({
+          name: 'FaceRecognition',
+          params: {kycStep: 2, kycStepBlocked: errorData.kycStepBlocked},
```
Comment (ice-ares) [inline src/store/modules/Tokenomics/sagas/startMiningSession.ts]: why would this be 2? selfie is 1

File: src/navigation/Main.tsx [hunk 2]
```diff
@@ -102,7 +103,10 @@ export type MainStackParamList = {
     targetCircleSize?: number;
     descriptionOffset?: number;
   };
-  FaceRecognition: undefined;
+  FaceRecognition: {
+    kycStep: FaceAuthKycNumber;
+    kycStepBlocked?: number;
```
Comment (ice-ares) [inline src/navigation/Main.tsx]: should also be FaceAuthKycNumber

File: src/store/modules/Tokenomics/sagas/startMiningSession.ts (hunk 1 above)
Comment (ice-hades) [reply src/store/modules/Tokenomics/sagas/startMiningSession.ts]: fixed

File: src/navigation/Main.tsx (hunk 2 above)
Comment (ice-hades) [reply src/navigation/Main.tsx]: fixed

File: src/store/modules/Tokenomics/sagas/startMiningSession.ts [hunk 3]
```diff
@@ -125,6 +103,34 @@ export function* startMiningSessionSaga(
...
+      const errorData = error?.response?.data?.data;
+      if (
+        errorData &&
+        typeof errorData.kycStepBlocked === 'number' &&
+        (errorData.kycStepBlocked === 1 || errorData.kycStepBlocked === 2)
+      ) {
+        yield removeScreenByName('Tooltip').catch();
+        navigate({
+          name: 'FaceRecognition',
+          params: {kycStep: 1, kycStepBlocked: errorData.kycStepBlocked},
```
Comment (ice-ares) [inline src/store/modules/Tokenomics/sagas/startMiningSession.ts]: kycStep: errorData.kycStepBlocked. To keep it consistent with line 125

File: src/store/modules/Tokenomics/sagas/startMiningSession.ts [hunk 4]
```diff
@@ -125,6 +103,33 @@ export function* startMiningSessionSaga(
...
+      }
+    } else if (isApiError(error, 403, 'MINING_DISABLED')) {
+      const errorData = error?.response?.data?.data;
+      if (errorData && typeof errorData.kycStepBlocked === 'number') {
+        if (errorData.kycStepBlocked === 1 || errorData.kycStepBlocked === 2) {
+          yield removeScreenByName('Tooltip').catch();
+          navigate({
+            name: 'FaceRecognition',
+            params: {
+              kycSteps: [1],
```
Comment (ice-ares) [inline src/store/modules/Tokenomics/sagas/startMiningSession.ts]: [errData.kycStepBlocked]

Comment (ice-hades) [reply src/store/modules/Tokenomics/sagas/startMiningSession.ts]: fixed

**Review by ice-ares — DISMISSED** [review]

**Review by ice-hector — DISMISSED** [review]

File: src/utils/ffmpeg.ts [hunk 5]
```diff
@@ -96,11 +133,9 @@ export async function getVideoDimensionsWithFFmpeg(videoUri: string) {
       const height = parseInt(match[1], 10);
       const width = parseInt(match[2], 10);
       return {width, height};
-    } else {
-      throw new Error('Failed to extract video dimensions.');
     }
   } catch (error) {
-    logError(error, {output});
-    throw error;
+    logError(error);
```
Comment (ice-orion) [inline src/utils/ffmpeg.ts]: Why you reverted logging ffmpeg output here? There are sentry reports regarding this case and it's unclear what is happening there without this info.

File: src/utils/ffmpeg.ts [hunk 6]
```diff
@@ -96,11 +133,9 @@ export async function getVideoDimensionsWithFFmpeg(videoUri: string) {
...
   }
+  return qualityToDimensions(VIDEO_QUALITY);
```
Comment (ice-orion) [inline src/utils/ffmpeg.ts:138]: Do we need to use the ffmpeg call at all if we can use `qualityToDimensions` for all the cases?

File: src/utils/ffmpeg.ts (hunk 5 above)
Comment (ice-hades) [reply src/utils/ffmpeg.ts]: because now we don't throw it and return defaults instead.

Comment (ice-orion) [reply src/utils/ffmpeg.ts]: yea but the error will still be logged and we won't be able to do anything with it

Comment (ice-orion) [reply src/utils/ffmpeg.ts]: se let's either remove the logging or add the debug info

Comment (ice-hades) [reply src/utils/ffmpeg.ts]: will remove

File: src/utils/ffmpeg.ts (hunk 6 above)
Comment (ice-hades) [reply src/utils/ffmpeg.ts:138]: I thought about this and it is risky. We don't know will those defaults fit all the devices of our users. Better keep it as a fallback only. And moreover I modified it so do this ffmpeg call only once for every emotions flow

Comment (ice-orion) [reply src/utils/ffmpeg.ts:138]: okay

File: src/utils/ffmpeg.ts (hunk 5 above)
Comment (ice-hades) [reply src/utils/ffmpeg.ts]: fixed

**Review by ice-orion — APPROVED** [review]

**Review by ice-ares — APPROVED** [review]

**Review by ice-hector — APPROVED** [review]

**Pull request merged**